lib/improver/tests/blending/spatial_weights/test_SpatialWeightsForMissingData.py

```diff
@@ -0,0 +1,787 @@
+        thresholds = [10]
+        data = np.ones((1, 2, 3), dtype=np.float32)
+        cycle1 = set_up_probability_cube(data, thresholds,
+                                         time=datetime(2017, 11, 10, 4, 0),
+                                         frt=datetime(2017, 11, 10, 0, 0),)
+        cycle2 = set_up_probability_cube(data, thresholds,
+                                         time=datetime(2017, 11, 10, 4, 0),
+                                         frt=datetime(2017, 11, 10, 1, 0),)
+        cycle3 = set_up_probability_cube(data, thresholds,
+                                         time=datetime(2017, 11, 10, 4, 0),
+                                         frt=datetime(2017, 11, 10, 2, 0),)
+        self.cube = CubeList([cycle1, cycle2, cycle3]).merge_cube()
+        self.cube = squeeze(self.cube)
+        self.plugin = SpatialWeightsForMissingData()
+
+    def test_no_mask(self):
+        """Test what happens when no mask is on the input cube"""
+        message = ("Input cube to SpatialWeightsForMissingData "
+                   "must be masked")
+        with self.assertRaisesRegex(ValueError, message):
+            self.plugin.create_initial_weights_from_mask(self.cube)
+
+    def test_basic(self):
+        """Test the weights coming out of a simple masked cube."""
+        input_data = np.array([[[10, 5, 10],
+                                [10, 5, 10]],
+                               [[5, 5, 10],
+                                [5, 5, 10]],
+                               [[10, 10, 10],
+                                [10, 10, 10]]],
+                              dtype=np.float32)
+        input_data = np.ma.masked_equal(input_data, 5)
+        self.cube.data = input_data
+        expected = np.array([[[1, 0, 1],
+                              [1, 0, 1]],
+                             [[0, 0, 1],
+                              [0, 0, 1]],
+                             [[1, 1, 1],
+                              [1, 1, 1]]],
+                            dtype=np.float32)
+        result = self.plugin.create_initial_weights_from_mask(self.cube)
+        self.assertArrayEqual(result.data, expected)
+        self.assertEqual(result.dtype, np.float32)
+        self.assertEqual(result.name(), "weights")
+
+    def test_none_masked(self):
+        """Test the weights coming out of a simple masked cube."""
+        input_data = np.array([[[10, 5, 10],
+                                [10, 5, 10]],
+                               [[5, 5, 10],
+                                [5, 5, 10]],
+                               [[10, 10, 10],
+                                [10, 10, 10]]],
+                              dtype=np.float32)
+        # This actually produces an array which numpy classes as NOT a masked
+        # array.
+        input_data = np.ma.masked_equal(input_data, 0)
+        self.cube.data = input_data
+        message = ("Input cube to SpatialWeightsForMissingData "
+                   "must be masked")
```

Comment:
No, if you print out the two input arrays they are different. The array in `test_no_mask` has no `.mask` attribute, but the one in `test_none_masked` does, although it is just set to `False` not an array of False. numpy seems to class both as not masked arrays. I got caught out when I originally wrote unit tests because I was testing whether the input array had a `.mask` attribute, not using the numpy is masked functionality.

I've modified the docstring.